xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -714,7 +714,7 @@ bool CDecoder::Open(AVCodecContext* avctx, AVCodecContext* mainctx, const enum A
   else if (avctx->codec_id == AV_CODEC_ID_VP9)
     m_vaapiConfig.maxReferences = 8;
   else if (avctx->codec_id == AV_CODEC_ID_AV1)
-    m_vaapiConfig.maxReferences = 18;
+    m_vaapiConfig.maxReferences = 20;
```

Comment:
That fix is not good. It will demand lots of memory from the decoder. Please detect those files dynamically, see H264 case. Especially if it is yet another Scene specific Anime nonsense ...

Reply:
See also: https://arxiv.org/pdf/2008.06091.pdf

Reply:
What do you mean by "those files"? This file is Main@L4.0 AV1 profile, it is nothing special.

Reply:
I linked the spec. It can only manage 8 reference frames and additional kodi already adds 6 more when opening to workaround broken profiles. holding for example 26 4K  reference frames in memory ... to workaround a single file is nonsense. This will for sure break some lowend setups.

A file with Main@L4.0 is nothing special, right - but that specific files requires to have 26 reference frames allocated? Quite special, isn't it?

Reply:
See also here: https://github.com/cconcolato/av1-spec/blob/master/05.decoding.overview.md

Would be good to find out what the user did to encode this file, which scene created them, etc.

Reply:
Filmgrain requires more references. When I did the initial implementation one of my testsamples had filmgrain and had pixelation issues when I lowered the maxreferences, that's how we came up with the value (and probably because the intel guys used a similar/same value in ffmpeg).

Reply:
Then let's use the same value used in ffmpeg, but checking for filmgrain before doing that. Sounds like a plan?

Reply:
/**
     * For film grain case, VAAPI generate 2 output for each frame,
     * current_frame will not apply film grain, and will be used for
     * references for next frames. Maintain the reference list without
     * applying film grain here. And current_display_picture will be
     * used to apply film grain and push to downstream.
    */

Yeah, so they need twice the references internally and only return the ones with filmgrain applied to outside. The filmgrain information should already be available in the private data of the avctx we are getting when opening?

Reply:
AV1DecContext *s = avctx->priv_data;
    const AV1RawFilmGrainParams *film_grain = &s->cur_frame.film_grain;
   int apply_grain = !(avctx->export_side_data & AV_CODEC_EXPORT_DATA_FILM_GRAIN) && film_grain- 
  >apply_grain;
  
  if (apply_grain) -> 20, else 10? or better said: 8 / 16?

Reply:
Might be doable in our demuxer as well via: AV_FRAME_DATA_FILM_GRAIN_PARAMS side data. 

Btw. if you both think that it's not worth the effort as only modern VAAPI devices can play AV1 hw decoded anyways and that 100 MB reference frame memory more for the worst case is not an issue -> please tell me.

Reply:
I tested some more samples with film grain, 4K and 1080p - http://download.opencontent.netflix.com/?prefix=AV1/Sparks/
No pixelation with those videos with current defaults. 
I'm not so sure that av1 film grain should require special handling/odd workarounds in Kodi. 
I see your concern regarding memory usage. However, as you pointed out, we are talking about modern GPUs/APUs with a hardware AV1 decoder onboard, so a bit more reference frame memory would not be an issue here.

Reply:
Great - then I hand over this PR to @a1rwulf who implemented the original AV1 handling. If he is fine I am also fine.